src/content/docs/en/guides/upgrade-to/v5.mdx

```diff
@@ -0,0 +1,1048 @@
+
+2. **Edit the collection definition**. Your updated collection requires a `loader`, and the option to select a collection `type` is no longer available.
+
+    ```ts ins={3,8} del={7}
+    // src/content/config.ts
+    import { defineCollection, z } from 'astro:content';
+    import { glob } from 'astro/loaders';
+
+    const blog = defineCollection({
+      // For content layer you no longer define a `type`
+      type: 'content',
+      loader: glob({ pattern: '**\/[^_]*.md', base: "./src/data/blog" }),
+      schema: z.object({
+        title: z.string(),
+        description: z.string(),
+        pubDate: z.coerce.date(),
+        updatedDate: z.coerce.date().optional(),
+      }),
+    });
+    ```
+
+3. **Change references from `slug` to `id`**. Content layer collections do not have a `slug` field. Instead, all updated collections will have an `id`.
+
+    ```astro ins={7} del={6}
+    // src/pages/[slug].astro
+    ---
+    export async function getStaticPaths() {
+      const posts = await getCollection('blog');
+      return posts.map((post) => ({
+        params: { slug: post.slug },
+        params: { slug: post.id },
+        props: post,
+      }));
+    }
+    ---
+    ```
+
+4. **Switch to the new `render()` function**. Entries no longer have a `render()` method, as they are now serializable plain objects. Instead, import the `render()` function from `astro:content`.
+
+    ```astro title="src/pages/index.astro" ins=", render" del={6} ins={7}
+    ---
+    import { getEntry, render } from 'astro:content';
+
+    const post = await getEntry('blog', params.slug);
+
+    const { Content, headings } = await post.render();
+    const { Content, headings } = await render(post);
+    ---
+    <Content />
+    ```
+</Steps>
+
+</details>
+
+## Deprecated
+
+The following deprecated features are no longer supported and are no longer documented. Please update your project accordingly.
+
+Some deprecated features may temporarily continue to function until they are completely removed. Others may silently have no effect, or throw an error prompting you to update your code.
+
+### Deprecated: `Astro.glob()`
+
+<SourcePR number="11826" title="Deprecate glob"/>
+
+In Astro v4.x, you could use `Astro.glob()` in your `.astro` components to query multiple files in your project. This had some limitations (where it could be used, performance, etc.), and using querying functions from the Content Collections API or Vite's own `import.meta.glob()` often provided more function and flexibility.
+
+Astro 5.0 deprecates `Astro.glob()` in favor of using `getCollection()` to query your collections, and `import.meta.glob()` to query other source files in your project.
+
+#### What should I do?
+
+Replace all use of `Astro.glob()` with `import.meta.glob()`. In most cases, this should work as a direct substitution. You should not require any updates to your [glob patterns](/en/guides/imports/#glob-patterns).
```

Comment:
Hi, just a few observations on the `Astro.glob` deprecation.

1. If there's a `then()` after Astro.glob, the substitute won't work, because it doesn't return a promise.
```js
await Astro.glob('./content_blocks/*.astro').then((components) => {

// Error: Property 'then' does not exist on type 'unknown[]'. ts(2339)
Object.values(import.meta.glob('./content_blocks/*.astro', { eager: true })).then((components) => {
```

I tried to figure out a better substitute that would solve this but I couldn't. I had to restructure my code.

2. In the docs, there's a lot of awaiting eager `import.meta.glob`, but it's not awaitable?

https://github.com/withastro/docs/blob/43691c97e2c24e7dc71b3be8f4c288c4e95ab369/src/content/docs/en/guides/markdown-content.mdx?plain=1#L222

```js
// Info: 'await' has no effect on the type of this expression. ts(80007)
await import.meta.glob('../pages/post/*.md', { eager: true });
```

It works anyways, but again, the new return type is not a promise, so it doesn't make sense to await it?

Just a few things I noticed. Maybe using Astro.glob is rare, so these things won't be a big problem for users.

Reply:
Thank you, @swift502 ! I'm checking with the team on these points to see what we want to show in the guide! Really appreciate the review comment!  🙌

Reply:
I guess this would be a more compatible substitute?

```js
(async () => Object.values(import.meta.glob('../../content_blocks/*.astro', { eager: true })))().then((components) => {
```

It's getting pretty ridiculous though. 😄

Reply:
Just updated the section now with some standard advice, and corrected examples! (We never really did show any chaining of `Astro.glob()` in the docs before, so we're keeping the example for migration super simple, so it's much easier to see the basic upgrade.

Thank you for being a part of our beta release!

Reply:
Sure, no problem. 🙂